From 226a2811faed5beb92c4ae2a58c425fe4dff5a6d Mon Sep 17 00:00:00 2001
From: Maarten De Decker <maarten.dedecker@mind.be>
Date: Tue, 27 May 2025 17:50:03 +0200
Subject: [PATCH] CI: bump prplOS version to June 5 2025

Bumps the prplOS version used in CI for regular and fast track platforms to f578eef4.
This fixes an issue where builds in prplMesh CI are failing because a checksum mismatch. (PPW-570)

Also upsteps easymesh_cert and the OSP builder version to allow nighly CI runs on OSPv1 and OSPv2.

Adds additional tags to platform builds in order to separate them across predefined buildservers.

Signed-off-by: Maarten De Decker <maarten.dedecker@mind.be>
---
 ci/easymesh_cert_version              |  2 +-
 ci/platform.yml                       | 23 +++++++++++++++++++----
 tools/docker/builder/openwrt/build.sh | 10 +++++-----
 3 files changed, 25 insertions(+), 10 deletions(-)

diff --git a/ci/easymesh_cert_version b/ci/easymesh_cert_version
index 09a06f06f1..93534f7f63 100644
--- a/ci/easymesh_cert_version
+++ b/ci/easymesh_cert_version
@@ -1 +1 @@
-1c1ff39a8160408a426daf773bf7f4cecc936a1d
+68c0e0e8265ce94cdfaf718563ad79cd2f90b2fb
diff --git a/ci/platform.yml b/ci/platform.yml
index a87a7d8df0..127e75be17 100644
--- a/ci/platform.yml
+++ b/ci/platform.yml
@@ -45,16 +45,28 @@ build-for-turris-omnia:
   extends: .build-for-openwrt
   variables:
     TARGET_DEVICE: "turris-omnia"
+  tags:
+    - shell
+    - docker-build
+    - turris-build
 
 build-for-haze:
   extends: .build-for-openwrt
   variables:
     TARGET_DEVICE: "haze"
+  tags:
+    - shell
+    - docker-build
+    - haze-build
 
 build-for-freedom:
   extends: .build-for-openwrt
   variables:
     TARGET_DEVICE: "freedom"
+  tags:
+    - shell
+    - docker-build
+    - freedom-build
 
 build-for-osp-ospv2:
   stage: build
@@ -67,8 +79,9 @@ build-for-osp-ospv2:
     - |
       trigger_details=$(curl -X POST --retry 3 -s -f -F \
       token=${OSP_PIPELINE_TRIGGER_TOKEN_2} \
-      -F ref="mainline-23.05-v3" \
+      -F ref="mainline-23.05-v4" \
       -F variables[PRPLMESH_REVISION]="$CI_COMMIT_SHA" \
+      -F variables[PRPLMESH_COMMIT_REF_NAME]="$CI_COMMIT_REF_NAME" \
       -F variables[PARENT_PIPELINE_ID]="$CI_PIPELINE_ID" \
       -F variables[PARENT_PROJECT_ID]="$CI_PROJECT_ID" \
       -F variables[MR_TARGET_SHA]="$CI_MERGE_REQUEST_TARGET_BRANCH_SHA" \
@@ -105,7 +118,7 @@ build-for-osp-ospv2:
       done
     - echo "OSP build succeeded! Visit pipeline at ${downstream_pipeline_url}"
   needs: []
-  timeout: 3h
+  timeout: 5h
   tags:
     - osp-build-fetch
 
@@ -226,6 +239,7 @@ run-certification-tests-R1-platform-essensium:
     - !reference [.run-certification-tests-R1, before_script]
     - echo "$TESTS_TO_RUN_ESSENSIUM" > .tests-to-run
     - echo "$DEVICE_UNDER_TEST_ESSENSIUM" > .device-under-test
+  timeout: 14h
   tags:
     - certs-tests-essensium
   needs:
@@ -246,6 +260,7 @@ run-certification-tests-R2-platform-essensium:
     - if: '$TESTS_TO_RUN_R2_ESSENSIUM !~ /^\$/ && $DEVICE_UNDER_TEST_R2_ESSENSIUM != "prplmesh"'
     - when: manual
       allow_failure: true
+  timeout: 14h
   tags:
     - certs-tests-R2-essensium
   needs:
@@ -261,11 +276,11 @@ run-certification-tests-R4-platform-essensium:
     - if: '$TESTS_TO_RUN_R4_ESSENSIUM !~ /^\$/ && $DEVICE_UNDER_TEST_R4_ESSENSIUM != "prplmesh" && $DEVICE_UNDER_TEST_R4_ESSENSIUM != "urx_osp" && $DEVICE_UNDER_TEST_R4_ESSENSIUM != "urx_ospv2"'
     - when: manual
       allow_failure: true
+  timeout: 14h
   tags:
     - certs-tests-R4-essensium
   needs:
-    - job: build-for-turris-omnia
-    - job: build-for-haze
+    - job: build-for-freedom
 
 changelog:
   stage: build
diff --git a/tools/docker/builder/openwrt/build.sh b/tools/docker/builder/openwrt/build.sh
index 1d0fd264bc..561d02048a 100755
--- a/tools/docker/builder/openwrt/build.sh
+++ b/tools/docker/builder/openwrt/build.sh
@@ -160,9 +160,9 @@ main() {
         OPENWRT_TOOLCHAIN_VERSION='21169344e223c5e02e8afedc3cc5648acd42f6cc'
         OPENWRT_VERSION='21169344e223c5e02e8afedc3cc5648acd42f6cc'
     else
-        dbg "$TARGET_DEVICE non-legacy platform, building on prplos mainline-23.05 + pWHM sah-next fixes"
-        OPENWRT_TOOLCHAIN_VERSION='320aee00e8aaac213b9e672420c7a74e1dc0cda2'
-        OPENWRT_VERSION='320aee00e8aaac213b9e672420c7a74e1dc0cda2'
+        dbg "$TARGET_DEVICE non-legacy platform, building on prplos mainline-23.05"
+        OPENWRT_TOOLCHAIN_VERSION='f578eef4cc38500cd8511031e782e1c71fb4a83b'
+        OPENWRT_VERSION='f578eef4cc38500cd8511031e782e1c71fb4a83b'
     fi
 
     dbg "OPENWRT_REPOSITORY=$OPENWRT_REPOSITORY"
@@ -202,8 +202,8 @@ VERBOSE=false
 IMAGE_ONLY=false
 OPENWRT_REPOSITORY='https://gitlab.com/prpl-foundation/prplos/prplos.git'
 # prplos mainline-23.05
-OPENWRT_TOOLCHAIN_VERSION='320aee00e8aaac213b9e672420c7a74e1dc0cda2'
-OPENWRT_VERSION='320aee00e8aaac213b9e672420c7a74e1dc0cda2'
+OPENWRT_TOOLCHAIN_VERSION='f578eef4cc38500cd8511031e782e1c71fb4a83b'
+OPENWRT_VERSION='f578eef4cc38500cd8511031e782e1c71fb4a83b'
 PRPLMESH_VARIANT="-nl80211"
 DOCKER_TARGET_STAGE="prplmesh-builder"
 SHELL_ONLY=false
-- 
GitLab

